### PR TITLE
fix: harden guardrails rules config shape

### PR DESCRIPTION
**Pull request by cloudenochcsis**
Fixes #

#### Describe the changes you have made in this PR -

- Treat malformed guardrails configs whose `rules` value is not a YAML list as invalid instead of iterating them.
- Add regression coverage for scalar, string, and mapping `rules` values returning no loaded rules.

### Demo/Screenshot for feature changes and bug fixes - 

```console
$ uv run ruff check app/guardrails/rules.py tests/test_guardrails/test_rules.py
All checks passed!

$ uv run ruff format --check app/guardrails/rules.py tests/test_guardrails/test_rules.py
2 files already formatted

$ uv run python -m pytest tests/test_guardrails -q
111 passed in 4.75s
```

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

`load_rules()` already treats missing or malformed guardrails config as no valid rules. This change extends that behavior to malformed `rules` values by validating that `rules` is a YAML list before iterating it. That keeps malformed scalar or mapping values from raising `TypeError` and preserves the existing behavior of returning an empty rule set for invalid config.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

File: tests/test_guardrails/test_rules.py [hunk 1]
```diff
@@ -26,6 +27,15 @@ def test_returns_empty_when_rules_key_missing(self, tmp_path: Path) -> None:
         path = _write_config(tmp_path, {"version": 1})
         assert load_rules(path) == []
 
+    @pytest.mark.parametrize("rules_value", [1, "secret", {"name": "bad"}])
```
[the remark this answers is not in the document]
Comment (cloudenochcsis) [reply tests/test_guardrails/test_rules.py]: Addressed in 0c9f6f04 by adding None to the malformed rules-value parametrization.

**Comment by cloudenochcsis** [comment]
@greptile review

**Comment by muddlebee** [comment]
not a pressing change, closing this
